### PR TITLE
[SUP-184] Registration timeout Fix &  CSDK 8, 9

### PR DESCRIPTION
**Pull request by chandrasekhar-plivo**
https://plivo-team.atlassian.net/browse/CSDK-9
https://plivo-team.atlassian.net/browse/CSDK-8
https://plivo-team.atlassian.net/browse/SUP-184

### CODE REVIEW

File: app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java [hunk 1]
```diff
@@ -91,7 +98,23 @@ private void registerReceiver() {
...
+        this.registerReceiver(networkReceiver, intentFilter);
+    }
 
+    protected void unregisterNetworkChanges() {
+        System.out.print("Unregistering network receiver");
+        try {
+            System.out.print("Unregistering network receiver 2");
+            unregisterReceiver(networkReceiver);
+        } catch (IllegalArgumentException e) {
+            System.out.print("exception ----- ");
```
Comment (Altanaib) [inline app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java]: remove

File: app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java [hunk 2]
```diff
@@ -91,7 +98,23 @@ private void registerReceiver() {
...
+    private void registerNetworkReceiver() {
+        IntentFilter intentFilter = new IntentFilter();
+        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION);
+        this.registerReceiver(networkReceiver, intentFilter);
+    }
 
+    protected void unregisterNetworkChanges() {
+        System.out.print("Unregistering network receiver");
+        try {
+            System.out.print("Unregistering network receiver 2");
```
Comment (Altanaib) [inline app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java]: change trace

File: app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java [hunk 3]
```diff
@@ -251,7 +273,7 @@ public void onReceive(Context context, Intent intent) {
     }
 
     private void handleIncomingCallIntent(Intent intent) {
-
+        System.out.print("Incoming call intent...");
```
Comment (Altanaib) [inline app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java]: remove

File: app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java [hunk 4]
```diff
@@ -464,7 +494,8 @@ public void run() {
     protected void onDestroy()
     {
         super.onDestroy();
-
+        System.out.print("On Destroy called");
```
Comment (Altanaib) [inline app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java]: remove / change with log

File: app/src/main/java/com/plivo/voicecalling/Helpers/Phone.java [hunk 5]
```diff
@@ -33,11 +33,12 @@ public static Phone getInstance(EndPointListner endPointListener) {
     private Phone() {
 
         endpoint = new Endpoint(true, this);
+        endpoint.setRegTimeout(479);
```
Comment (Altanaib) [inline app/src/main/java/com/plivo/voicecalling/Helpers/Phone.java]: add comment / description for 8 min timeout

File: app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java (hunk 2 above)
Comment (chandrasekhar-plivo) [reply app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java]: done

File: app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java (hunk 3 above)
Comment (chandrasekhar-plivo) [reply app/src/main/java/com/plivo/voicecalling/Activities/VoiceActivity.java]: done

File: app/src/main/java/com/plivo/voicecalling/Helpers/Phone.java (hunk 5 above)
Comment (chandrasekhar-plivo) [reply app/src/main/java/com/plivo/voicecalling/Helpers/Phone.java]: done

**Review by Altanaib — COMMENTED** [review]
ok. To be approved after approving SDK

**Pull request merged**